src/dagmc/DagMC.cpp

```diff
@@ -0,0 +1,774 @@
+ErrorCode DagMC::closest_to_location(EntityHandle volume,
+                                     const double coords[3], double& result,
+                                     EntityHandle* surface) {
+  ErrorCode rval = GQT->closest_to_location(volume, coords, result, surface);
+  return rval;
+}
+
+// calculate volume of polyhedron
+ErrorCode DagMC::measure_volume(EntityHandle volume, double& result) {
+  ErrorCode rval = GQT->measure_volume(volume, result);
+  return rval;
+}
+
+// sum area of elements in surface
+ErrorCode DagMC::measure_area(EntityHandle surface, double& result) {
+  ErrorCode rval = GQT->measure_area(surface, result);
+  return rval;
+}
+
+// get sense of surface(s) wrt volume
+ErrorCode DagMC::surface_sense(EntityHandle volume, int num_surfaces,
+                               const EntityHandle* surfaces, int* senses_out) {
+  ErrorCode rval = GTT->get_surface_senses(volume, num_surfaces, surfaces,
+                                           senses_out);
+  return rval;
+}
+
+// get sense of surface(s) wrt volume
+ErrorCode DagMC::surface_sense(EntityHandle volume, EntityHandle surface,
+                               int& sense_out) {
+  ErrorCode rval = GTT->get_sense(surface, volume, sense_out);
+  return rval;
+}
+
+ErrorCode DagMC::get_angle(EntityHandle surf, const double in_pt[3],
+                           double angle[3],
+                           const RayHistory* history) {
+  ErrorCode rval = GQT->get_normal(surf, in_pt, angle, history);
+  return rval;
+}
+
+ErrorCode DagMC::next_vol(EntityHandle surface, EntityHandle old_volume,
+                          EntityHandle& new_volume) {
+  ErrorCode rval = GTT->next_vol(surface, old_volume, new_volume);
+  return rval;
+}
+
+/* SECTION III */
+
+EntityHandle DagMC::entity_by_id(int dimension, int id) {
+  return GTT->entity_by_id(dimension, id);
+}
+
+int DagMC::id_by_index(int dimension, int index) {
+  EntityHandle h = entity_by_index(dimension, index);
+  if (!h)
+    return 0;
+
+  int result = 0;
+  MBI->tag_get_data(GTT->get_gid_tag(), &h, 1, &result);
+  return result;
+}
+
+int DagMC::get_entity_id(EntityHandle this_ent) {
+  return GTT->global_id(this_ent);
+}
+
+ErrorCode DagMC::build_indices(Range& surfs, Range& vols) {
+  ErrorCode rval = MB_SUCCESS;
+
+  // surf/vol offsets are just first handles
+  setOffset = std::min(*surfs.begin(), *vols.begin());
+
+  // max
+  EntityHandle tmp_offset = std::max(surfs.back(), vols.back());
+
+  // set size
+  entIndices.resize(tmp_offset - setOffset + 1);
+
+  // store surf/vol handles lists (surf/vol by index) and
+  // index by handle lists
+  surf_handles().resize(surfs.size() + 1);
+  std::vector<EntityHandle>::iterator iter = surf_handles().begin();
+  *(iter++) = 0;
+  std::copy(surfs.begin(), surfs.end(), iter);
+  int idx = 1;
+  for (Range::iterator rit = surfs.begin(); rit != surfs.end(); ++rit)
+    entIndices[*rit - setOffset] = idx++;
+
+  vol_handles().resize(vols.size() + 1);
```

Comment:
Rather than explicitly pulling the `iterator`, setting it to `0`, and incrementing it, it might be more readable to use the technique shown with `group_handles()` on line 409.  E.g.:

```
vol_handles().resize(vols.size() + 1);
vol_handles()[0] = 0;
std::copy(vols.begin(), vols.end(), &vol_handles()[1]);
```